this is a merge from the richfaces quickstart project: kitchensink-rf

bleathem opened the pull request.

jfuerth committed:
New quickstart: Errai Hello World

Now with fixes for some of the issues identified by Pete Muir in the pull request.

tomjenkinson committed:
Initial version of distributed crashrec

mmusgrov committed:
JTA transaction crash recovery example

Commit:
added tasks-jsf sample (quickstart #10), based on tasks (quickstart #8)

Commit:
Added SLSB tempconvert project

Commit:
Added quickstart #8 - tasks

spagop committed:
ejb 3.1 stateful session bean quickstart

Commit:
Fixes based on Pete's review

pmuir committed:
Use SFSB with XPC injection

jtosi committed:
Add mail

pmuir committed:
Make Resources request scoped

Commit:
Reformated, updated dependencies to 7.1.0.Final

Commit:
initial commit of quickstart-28: richfaces-validation (based upon quickstart-27)

Commit:
Added information about server log contents to README

andytaylor committed:
added support for Openshift in helloworld-mdb

dmison committed:
Logging tools quickstart

Signed-off-by: Darrin Mison <dmison@me.com>

pmuir committed:
Tweak openshift process

Commit:
Added Arquillian tests to ejbinwar

baranowb committed:
add block comments, fix how error is displayed

stliu committed:
change to use jboss-javaee-6.0-with-tools bom

Commit:
Major re-work of JMS quickstart

sgilda committed:
Add info to run add-user script. Remove all tables but the one describing the quickstarts

pmuir committed:
Minor fixes

fjuma committed:
JBPAPP-8403: Updated the instructions for starting the server when using EAP 6 in the README.md files for cmt, helloworld-mdb, and jts. Also fixed spacing in the jts/README.md file so that the code blocks in the generated html files will be rendered properly without escaping the tags.

Commit:
Added EJB Security

Commit:
Added Servlet security

Commit:
Changed user name and password from admin/admin to UserA/password

darranl committed:
Remove dependency on specific version and set the local versions.

OndraZizka committed:
AS7-4130  AS 7 quickstart: Wicket EAR and Wicket WAR

OndraZizka committed:
AS7-4130  AS 7 quickstart: Wicket (WAR)

pmuir committed:
Rework xml-jaxp

* remove css and styling
* flatten packages
* use CDI better
* flatten out JSF

pmuir committed:
Various

* Flatten packages
* remove readme.html
* remove JBoss repo from POMN
* Add -ds.xml support
* Remove synthetic ID from hash code and equals
* Add unique column on username
* use identity based ids
* use positive numbers for ids
* Add stuff to README

pmuir committed:
Update root README and pom for new quickstarts

pmuir committed:
Regularise use of spaces, new lines etc, to give correct indent in code blocks

pmuir committed:
Fix wording and ws

mmusgrov committed:
Add instructions for deploying to OpenShift

paulrobinson committed:
Updated to work on Openshift

sgilda committed:
Template for creating a README.md file for new quickstarts

pmuir committed:
Various

* format README better and tidy up language
* Use EJB no interface view
* Use Servlet 3 annotations for security
* whitespace/formatting
* use guest role to match root readme

pmuir committed:
Various

* format README better and tidy up language
* Use Servlet 3 annotations for security
* whitespace/formatting
* use guest role to match root readme

pmuir committed:
Tidy up assembled zip

sgilda committed:
Add pom.xml file to template directory. Fix Apache License issues for existing pom.xml files.

sgilda committed:
Modify artifactId to use the replaceable QUICKSTART_NAME that is used everywhere else in the pom and README files.

jamezp committed:
Update the maven plugin versions

jamezp committed:
Add missing modules to parent pom.

sgilda committed:
Fix instructions to deploy EARs

pmuir committed:
Require Bash 3

sgilda committed:
Fix broken link to http://www.jboss.org/jbossas in default.xhtml files

pmuir committed:
Include EAP / JBDS instructions

pmuir committed:
Minor formatting and text

pmuir committed:
* Grammer, spelling
* Flatten packages
* Add deployable datasources
* Use index.html not index.jsp

pmuir committed:
* Fix whitespace
* Fix grammer and spelling
* Remove unneeded Serializable
* Remove serialUID
* Use index.html not index.jsp

paulrobinson committed:
minor fixes

pmuir committed:
Fix version in comment

pmuir committed:
* Use correct terminology in readme
* Use EJB Client BOM
* Move common dependencies to parent POM
* Minor work on server components
* Format comments on client code better

pmuir committed:
Updates to OpenShift instructions

* use jboss-as-quickstart account
* 4 space indent for code blocks
* minor text changes
* use random directory, not subdir of quickstart

pmuir committed:
Minor improvements to README based on Paul's readme

pmuir committed:
Improvements based on Mike's readme + ws

starksm64 committed:
Allow the java.naming.provider.url to be set externally

pmuir committed:
Improve HTML renderer

* Add stylesheet
* Add syntax highlight via Pygments

sgilda committed:
Updated the README instructions with the new property information plus other enhancements.

pmuir committed:
Whitespace

pmuir committed:
Updates based on archetype refresh

pmuir committed:
Various fixes to kitchensink and kitchensink-ear

* Apply rebrand to both
* Update kitchensink-ear to use the split controller/business logic from Brian
* Update to jboss-bom 1.0.0.M6
* Apply any updates from archetype regen

sgilda committed:
Updated the root and template README files. Updated quickstart list, fixed typos.

sgilda committed:
Remove JBOSS_HOME environment variable and point to root README for arquillian instructions

pmuir committed:
Less clunky sentence

pmuir committed:
Various work on wicket-ear

* Split out wicket-war from wicket-ear branches
* Remove whitespace
* Use standard groupId and artifactId
* Remove custom ear packaging
* Default is skip for jboss-as plugin
* Move all dependencyManagement to parent
* Add deployable datasources
* Generate application.xml
* Use JBoss Java EE APIs exclusively
* Remove redundant Local suffix from ContactDao
* Use fields for JPA annotations
* Don't use synthetic ID for equality and hash code, well known anti-pattern
* Use CDI not EJB injections into Wicket (promote modern approaches)
* Improve comments

pmuir committed:
Fix comment in kitchensink-ear -ds.xml

pmuir committed:
Various work on wicket-war

* Split out wicket-war from wicket-ear branches
* Remove whitespace
* Use standard groupId and artifactId
* Move all version control dependencyManagement
* Add deployable datasources
* Remove redundant Local suffix from ContactDao
* Use fields for JPA annotations
* Don't use synthetic ID for equality and hash code, well known anti-pattern
* Use CDI not EJB injections into Wicket (promote modern approaches)
* Improve comments

pmuir committed:
A few more comments for wicket-ear

sgilda committed:
Attempt to standardize README.md files.

jmesnil committed:
Reinstate queue deployment inside the Web archive

pmuir committed:
Minor fixes to release-utils

sgilda committed:
More readme cleanup

pmuir committed:
Various alignment and updates

* Use H2 Console quickstart
* Use memory database (H2 console quickstart allows this)
* Align DS naming with quickstart
* Make fields private
* Fix warnings
* Import JTA APIs

baranowb committed:
initial dom4j port of q40

baranowb committed:
remove deps on unused libs, clean pom, remove templates

pmuir committed:
Add dom4j xml example to root pom.

pmuir committed:
Whitespace

balunasj committed:
Updated CONTRIBUTING.md with rules related to variants of the core kitchensink quickstart

fjuma committed:
Update version of jboss-javaee-6.0 to 3.0.0.Final

fjuma committed:
Update to jboss-jsf-api_2.1_spec since jboss-javaee-6.0 was updated to 3.0.0.Final.

pmuir committed:
Whitespace & bump version

pmuir committed:
whitespace

sgilda committed:
Fix issues 170,171,172. Modify jta-crash-rec to follow template and change web form instructions

mbogoevici committed:
Update JAX-RS service scheme to full JSON

mbogoevici committed:
Clean up imports

mbogoevici committed:
Updated JAX-RS services to full JSON - for kitchensink-ear

tomjenkinson committed:
Update for persistence xml correction

tomjenkinson committed:
Updated the database name

baranowb committed:
update version

pmuir committed:
Split quickstarts into groups for easy testing.

pmuir committed:
Whitespace, fix quickstart names

pmuir committed:
Whitespace, disable JBoss AS plugin

pmuir committed:
Revert "Update for persistence xml correction"

This reverts commit bf81ff373eac14434e1627fc12f597ab1508d13c.

pmuir committed:
Revert "Updated the database name"

This reverts commit 3b654ad48d1ce6d0ae4e0e2d62532d69deff9699.

pmuir committed:
Fix Datasource name

pmuir committed:
Fix ws, Fix DS name, use back ticks for code

pmuir committed:
Align readme (thanks to Sande!)

pmuir committed:
Update BOM

pmuir committed:
Add wicket-war to root pom

dmison committed:
setLenient to false to enforce correct date formats

Signed-off-by: Darrin Mison <dmison@me.com>

baranowb committed:
JBOSSAS_186

pmuir committed:
Update to latest BOM versions

sgilda committed:
Add Maven configuration instructions for EAP6: issue 190 and JBPAPP-8293

tomjenkinson committed:
use separate databases for cmt and jts

sgilda committed:
Missed adding the arquillian instructions, fixed indentations, modified jta-crash-rec and jts-distributed-crash-rec, add missing authors

pmuir committed:
Solve issue #194 :-) auto transform links from README.md -> README.html

mbogoevici committed:
Separated producer from repository

- isolate list producer as a separate request-scoped bean
- made repository application-scoped

mbogoevici committed:
Fix: wrong exception caught - must be NoResultException

blainemincey committed:
Made modifications to include instructions on how to run this on the helloworld-rs application deployed to OpenShift.

blainemincey committed:
Add OpenShift support to jax-rs-client

* Added surefire plugin and system props to pom.xml.
* Read from system props in JaxRsClientTest.java.
* Added comment to the beforeClass method.
* Fix whitespace

pmuir committed:
Update artifactId and version

pmuir committed:
Fix test instructions

sgilda committed:
Update quickstart table. Clarify PostgreSQL configuration

pmuir committed:
Remove odd trailing slashes

sgilda committed:
issue 179 (JIRA RF-12102) - change connection-url for datasources

pmuir committed:
Prepare for 7.1.1.CR1 release

pmuir committed:
Prepare for development of 7.1.1-SNAPSHOT

pmuir committed:
Add note on upload to script

pmuir committed:
Expand out <a>

sgilda committed:
Fixed issues 201:Missing Appache licenses, 203: wrong versions, 206:jta-crash-rec README fixes, 207: Replace anchor tags to be compatible with markdown

bleathem committed:
helloworld-jsf: incremented the RichFaces version to 4.2.0.Final

mmusgrov committed:
JAX-RS Tasks Quickstart No 33 (and 34)

Commit:
Fixed issue with incorrect build status message as a result of a failure

sgilda committed:
Updates for added temperature-converter quickstart. Fix minor README issues.

mmusgrov committed:
Incorporate Sandes changes

jaikiran committed:
Fix variety of issues in ejb-remote: Remove unneeded JBossSaslProvider registration, fix client POM to use jboss-as-ejb-client-bom, Remove unused properties from jboss-ejb-client.properties file

sgilda committed:
Fix incorrect javaee6.with.tools.version properties for kitchensink-jsp, tasks, and tasks-jsf

madhumita committed:
modification to make hibernate4 in sync with kitchensink changes

sgilda committed:
Fix for JIRA JBPAPP-8681: need instructiosn to add JMS test queue for EAP6 server only

Commit:
Merging 1.0.0.M3b from aerogear/as-quickstarts

pgier committed:
[AS7-3887] Use the 7.1.1.Final version of the jboss-as-maven-plugin

paulrobinson committed:
Allow wsat-simple to use default context on openshift

sgilda committed:
Fix readme file issues: helloworld-jms warns of error message, mail had wrong paths and names for configuration files

pmuir committed:
JBPAPP-8867

pmuir committed:
Fixes #107 - clarify that helloworld-jms is a standalone, Java SE client for JMS

pmuir committed:
whitespace

pmuir committed:
License headers part 1

pmuir committed:
License header script

pmuir committed:
Fix issue #177, ordering of elements

pmuir committed:
Fix context-path to be jboss-as-wsat-simple

pmuir committed:
Add empty XML elements to prevent SAX errors

pmuir committed:
Update BOM versions

pmuir committed:
Resolve issue 205, remove JBoss repo definitions

* Additionally add needed Servlet dependency

pmuir committed:
Switch to new version of rhc commands

sgilda committed:
Issue 232 (JBPAPP-8893): Undeploy failure for jts. Modified undeploy command in README. Other minor README fixes

sgilda committed:
Fix README problems found in QE testing of EAP 6 quickstarts

sgilda committed:
Add caveat that only the as7 quickstarts currently run on OpenShift

pmuir committed:
Add AS config and conform to template

pmuir committed:
oops

sgilda committed:
clarify that the domain portion of the OpenShift URL will be your own domain

LightGuard committed:
Changing to /tasks

Also adding import.sql so the arquillian tests run and pass.

LightGuard committed:
Updating the README.

Sorry, didn't even look at the README, should have.

sgilda committed:
tasks-rs quickstart was out of order in the quickstart table

Commit:
Minor edits to HTML, and EJB (3 decimal places on return)

pmuir committed:
Remove unneeded steps

pmuir committed:
Prepare for 7.1.1.CR2 release

pmuir committed:
Prepare for development of 7.1.1-SNAPSHOT

madhumita committed:
added/modified files to implement UI changes in sync with kitchensink

pmuir committed:
Add a nice index page for hellworld-rs :-)

pmuir committed:
Add empty beans element

pmuir committed:
Improve script thanks to Tobias Kunze

sgilda committed:
README Maven config updates, change header titles to quickstart folder name, update experience levels, issue 258

Modified level for bmt to match cmt

madhumita committed:
added fix to handle exception for unique member id

sgilda committed:
Fix error handling for duplicate key and other exceptions: display root cause

sgilda commented:
Hi Brian, I'm having a few problems with this one. 
Is this rebased off of  the current jbossas/quickstart master? 
When I checkout your branch, immediately I see a bunch of untracked README.html files (one for every quickstart in addition to the kitchensink-rf and root README file). We no longer check in the README.html files. They need to be removed.
Maybe I'm doing something wrong here?
Thanks, Sande

bleathem commented:
Yes, a git pull from upstream master indicates that everything is up-to-date.

I added a single readme.html file, the one you e-mailed me the other week, along with the markdown one.  Should I remove it?

If I look in the files listed in the pull request, I don't see anything outside of the kitchensink-rf folder.  Are you sure those readme.html files in the other quickstarts come from this pull request?

sgilda commented:
Apparently they don't come from your pull. It's very strange.
Yes, please remove the README.html file. I think I sent it to you so it 
would be easier view than the markdown.

BTW, the quickstart deployed and ran fine. :-)

On 05/15/2012 11:08 AM, Brian Leathem wrote:

> Yes, a git pull from upstream master indicates that everything is up-to-date.
> 
> I added a single readme.html file, the one you e-mailed me the other week, along with the markdown one.  Should I remove it?
> 
> If I look in the files listed in the pull request, I don't see anything outside of the kitchensink-rf folder.  Are you sure those readme.html files in the other quickstarts come from this pull request?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/262#issuecomment-5718249

sgilda commented:
Brian, let me know when you've removed the README.html file and I'll try this again.

bleathem commented:
I removed the README.html file

sgilda commented:
Pete or Brian, 
I started over. Brian, I updated to get your latest, then checked out a branch to get your stuff.

I'm not familiar enough with `git`, so I'm not sure what this list of README.html files means, but I don't think it's good.
Brian, you removed the kitchensink-rf.html file, so it no longer shows up. But the others are all still there.

Pete, is this going to be a problem on checkin?

The following is what I see when I do a `git status` after checking out the kitchensink-rf.

quickstart (master) $ git fetch bleathem
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (1/1), done.
remote: Total 3 (delta 2), reused 3 (delta 2)
Unpacking objects: 100% (3/3), done.
From https://github.com/bleathem/quickstart
   717d148..ef89f99  master     -> bleathem/master

quickstart (master) $ git checkout -b kitchensink-rf-bleathem bleathem/master
Branch kitchensink-rf-bleathem set up to track remote branch master from bleathem.
Switched to a new branch 'kitchensink-rf-bleathem'
quickstart (kitchensink-rf-bleathem) $ git status
`# On branch kitchensink-rf-bleathem`
`# Untracked files:`
`#   (use "git add <file>..." to include in what will be committed)`
`#`
`#  README.html`
`#  bean-validation/README.html`
`#  bmt/README.html`
`#  cdi-injection/README.html`
`#  cmt/README.html`
`#  ejb-in-ear/README.html`
`#  ejb-in-war/README.html`
`#  ejb-remote/README.html`
`#  ejb-security/README.html`
`#  forge-from-scratch/README.html`
`#  greeter/README.html`
`#  h2-console/README.html`
`#  helloworld-errai/README.html`
`#  helloworld-gwt/README.html`
`#  helloworld-html5/README.html`
`#  helloworld-jms/README.html`
`#  helloworld-jsf/README.html`
`#  helloworld-mdb/README.html`
`#  helloworld-osgi/README.html`
`#  helloworld-rs/README.html`
`#  helloworld-singleton/README.html`
`#  helloworld/README.html`
`#  hibernate3/README.html`
`#  hibernate4/README.html`
`#  jax-rs-client/README.html`
`#  jta-crash-rec/README.html`
`#  jts-distributed-crash-rec/README.html`
`#  jts/README.html`
`#  kitchensink-ear/README.html`
`#  kitchensink-html5-mobile/README.html`
`#  kitchensink-jsp/README.html`
`#  kitchensink/README.html`
`#  log4j/README.html`
`#  logging-tools/README.html`
`#  mail/README.html`
`#  numberguess/README.html`
`#  payment-cdi-event/README.html`
`#  richfaces-validation/README.html`
`#  servlet-async/README.html`
`#  servlet-filterlistener/README.html`
`#  servlet-security/README.html`
`#  shopping-cart/README.html`
`#  tasks-jsf/README.html`
`#  tasks-rs/README.html`
`#  tasks/README.html`
`#  temperature-converter/README.html`
`#  template/README.html`
`#  wicket-ear/README.html`
`#  wicket-war/README.html`
`#  wsat-simple/README.html`
`#  wsba-coordinator-completion-simple/README.html`
`#  wsba-participant-completion-simple/README.html`
`#  xml-dom4j/README.html`
`#  xml-jaxp/README.html`
nothing added to commit but untracked files present (use "git add" to track)

sgilda commented:
I should mention that the only files in the commit are richfaces-rf files.
I should also mention I tested it and it works great.

Pete, I'm good with this one if you are. I'm just not sure how to handle the README.html files other than delete them.

Commit:
Arquillian release upgrade

Commit:
java ee specs upgrade

Commit:
skip tests on minify

Commit:
bump to 1.0.0.M4 release

Commit:
Upgrading AS maven plugin

pmuir committed:
Move metadata to quickstarts

pmuir committed:
Move release process to CONTRIBUTING.md

pmuir committed:
Add auto-generated TOC for readmes

pmuir committed:
Clean up whitespace, imports and warnings

pmuir commented on kitchensink-rf/pom.xml:
```diff
@@ -0,0 +1,309 @@
+      <dependency>
+         <groupId>org.hibernate</groupId>
+         <artifactId>hibernate-validator</artifactId>
+         <version>4.2.0.Final</version>
```
This should be in the -with-hibernate bom I think?

bleathem replied:
This was "inherited" from the kitchensink project.  I'll review how the kitchensink pom has evolved since I forked it.

pmuir commented on kitchensink-rf/pom.xml:
```diff
@@ -0,0 +1,309 @@
+      <dependency>
+         <groupId>org.hibernate</groupId>
+         <artifactId>hibernate-jpamodelgen</artifactId>
+         <version>1.1.1.Final</version>
```
Should be in the -with-hibernate bom

bleathem replied:
Same as above re: inheriting from the kitchensink project

pmuir commented on kitchensink-rf/pom.xml:
```diff
@@ -0,0 +1,309 @@
+            <groupId>org.jboss.as.plugins</groupId>
+            <artifactId>jboss-as-maven-plugin</artifactId>
+         </plugin>
+         <plugin>
```
Comments?

pmuir commented on kitchensink-rf/src/main/java/org/jboss/as/quickstarts/kitchensinkrf/controller/PageBean.java:
```diff
@@ -0,0 +1,53 @@
+import java.io.Serializable;
+
+/**
+ * @author <a href="http://community.jboss.org/people/bleathem">Brian Leathem</a>
```
Comment needed here.

pmuir commented on kitchensink-rf/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -0,0 +1,26 @@
+  <!-- add a welcome-file-list entry to allow JSF pages to be sued as welcome files -->
+  <welcome-file-list>
+    <welcome-file>index.jsf</welcome-file>
+  </welcome-file-list>
```
I would prefer we stick to the index.html based redirect, in order to be consistent across all quickstarts

bleathem replied:
The html based re-direct is gross, as it requires a 2nd request to originate from the client. With the RichFaces mobile approach, we've got to great pains to reduce the number of request made by the client in order to improve the responsiveness of the application.  It's a shame to have to introduce an additional request just to accommodate a spec problem.

Can we instead look at getting the other quickstarts to use a index.jsf welcome page?

pmuir replied:
Sure, please send a pull.

bleathem replied:
Sorry for the delay - only getting to this now.

@pmuir if we go ahead an do this, it requires adding a web.xml to every project to define the welcome-file list.  Are you ok with introducing a web.xml file for this?

pmuir replied:
That kinda sucks :-(

pmuir commented on kitchensink-rf/src/main/webapp/index.xhtml:
```diff
@@ -0,0 +1,98 @@
+
+        <div>
+            <p>You have successfully deployed a RichFaces web application on JBoss AS 7.</p>
+
```
This isn't right. It should be:

```
<div>
     <p>You have successfully deployed a RichFaces & Java EE 6 web application.</p>
     <h3>Your application can run on:</h3>
     <img src="resources/gfx/dualbrand_as7eap.png"/>
  </div>
```

Or something like that. JBoss AS 7 shouldn't feature.

pmuir commented on kitchensink-rf/src/main/webapp/mobile/home.xhtml:
```diff
@@ -0,0 +1,35 @@
+         <div class="mobileContent">
+            <h1>Welcome to JBoss AS 7!</h1>
+            <p>
+               You have successfully deployed a RichFaces web application on JBoss AS 7, and are viewing it via your mobile
```
Same comments, here  - need to keep the brand consistent.

pmuir commented on kitchensink-rf/src/main/webapp/mobile/index.jsf:
```diff
@@ -0,0 +1,6 @@
+This empty file is required to "trick" the servlet container into beleiveing "index.jsf" exists, and can be used as a
+welcome file.  Serving of this resource will then instead be performed by the Faces Servlet.
+
+** Do not delete this file. **
```
Same as above.

sgilda committed:
JBPAPP-8961: Readme fixes for jta-crash-rec

snjeza committed:
Add quickstarts to JBoss Tools

wfink committed:
Initial version of quickstart cluster-ha-singleton-ejb

pmuir committed:
remove POH5 reference

pmuir committed:
Update to 1.0.0.Final BOM

pmuir committed:
Prepare for 7.1.1.Final release

wfink committed:
New quickstart for asynchronous EJB invocation

pmuir committed:
Add an inter-application communication quickstart

* Uses EJBs to communicate
* Aliases EJBs as CDI beans to hide details from
  clients of the beans
* Thanks to Jason Greene for part of description

pmuir committed:
Update ignores

pmuir committed:
Use OO programming and enums to simplify arch

pmuir committed:
Initial import of Getting Started Developing Applications Guide

* Convert to asciidoc and update to current quickstarts

pmuir committed:
Update ignores

pmuir committed:
Code simplification and whitespace

pmuir committed:
-s isn't available on all platforms, pipe through sort instead

sgilda committed:
Fix issue 295: Replace github URLs with jdf URL in parent POM, remove other URL, add hibernate3

sgilda committed:
Fix issue 289: Minor README updates that were fixed downstream in EAP6

paulrobinson committed:
jboss-jdf_296 wsat-simple uses obsolete github URL for namespaces

pmuir committed:
Resolve issues identified by Sande Gilda

sgilda commented:
I'm a little confused by the last few comments. It sounds like we should be changing all the other quickstarts to use the index.jsf file like this one. 

Am I understanding this correctly?
If that is true, is this one is ready to merge?

bleathem commented:
That's my understanding as well.  Pete has asked me to submit a pull request implementing such a change.  I intend to submit a pull request but had to put life on hold while prepping for and attending both JBW and JAX.  Those are behind me now, and I hope to get to this by the end of next week at the latest.

sgilda commented:
Thanks for the clarification. I just wanted to make sure this one was 
good to go. :-)

On 07/16/2012 02:40 PM, Brian Leathem wrote:

> That's my understanding as well.  Pete has asked me to submit a pull request implementing such a change.  I intend to submit a pull request but had to put life on hold while prepping for and attending both JBW and JAX.  Those are behind me now, and I hope to get to this by the end of next week at the latest.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jboss-jdf/jboss-as-quickstart/pull/262#issuecomment-7014987

pmuir commented:
My main concern is consistency, I want all quickstarts to do this in the same way. I don't want to end up with myriad different approaches, so I would prefer not to change one before the rest.

sgilda commented:
Makes sense. :-)

pmuir committed:
Add author to the guide

sgilda committed:
Fix issue #301 and issue #302 : Remove OpenShift warning, Make template README instructions clearer

sgilda committed:
Fix issue #280: add setContentType to doGet method in servlet and in guide

sgilda committed:
Fixed issue #270: Commented alternative version has wrong element name

pmuir committed:
Align/simplify naming

pmuir committed:
improve javadoc

sgilda committed:
Update URLs for the Get Started docs for both community and product

sgilda committed:
Fix issue #257: Add better instructions to greeter README

pmuir committed:
Update references to point to jdf

pmuir committed:
Add metadata for cluster-ha-singleton and inter-app

pmuir committed:
fix TOC generation

Closes #300

pmuir committed:
Remove duplicated/redundant info

pmuir committed:
Use Maven versions plugin to update versions

pmuir committed:
Fix version numbers

pmuir committed:
Prepare for 7.1.2.M1 release

pmuir committed:
Prepare for development of 7.1.2-SNAPSHOT

pmuir committed:
Update Archetype guide to recommend using JBDS

sgilda committed:
Fix #274: tasks README is missing the step to start the server

sgilda committed:
Fix #277: remove hibernate-validator and junit from kitchensink-ear parent POM.

sgilda committed:
Partial fix for issue #245: apply dual branding for hibernate3

sgilda committed:
Partial fix for issue #245: Dual branding for kitchensink-jsp

pmuir committed:
Remove test dependencies

fixes #269

pmuir committed:
Re-add dual brand

pmuir committed:
Partial fix for issue #245: More dual branding for kitchensink-jsp

pmuir committed:
Minor changes

* Update readme
* Use standard names

sgilda committed:
Partial fix for issue 287. Still waiting for BOMs

sgilda committed:
Fix OpenShift instructions, issue #291, developer and doc links.

dmison committed:
added index html page

Signed-off-by: Darrin Mison <dmison@me.com>

pmuir committed:
Update BOM comments

pmuir committed:
whitespace

dmison committed:
made the <a href>s relative so they always work, also moved examples to the top of each section

Signed-off-by: Darrin Mison <dmison@me.com>

dmison committed:
fixed indenting etc, as per https://github.com/jboss/ide-config

Signed-off-by: Darrin Mison <dmison@me.com>

dmison committed:
added reference to landing page to readme

Signed-off-by: Darrin Mison <dmison@me.com>

wfink committed:
Scheduler not removed within the HASingleton stop method

sgilda committed:
Add localized version of the kitchensink

LightGuard committed:
CDI Portable Extension Quickstart
Code complete for cdi-portable-extension

pom, template readme, beans.xml files missed by IntelliJ

Adding openshift profile

README done

LightGuard committed:
Removing openshift from pom, moving static block into a ctor.

jamezp committed:
Use properties for dependency and plugin versions.

jamezp committed:
Fix invalid rest URL's.

OndraZizka committed:
Unify module names style.

LightGuard committed:
Addressing comments made by Sande

Added a small section about PEs and what this one address in the README.
Using the new properties in the pom
Adding the class in the Extension services file (oops)

sgilda committed:
Add localized version of kitchensink-ear quickstart

sgilda committed:
Add dual brand and fix documentation and product links

sgilda committed:
Fix product documentation link in template jsf

OndraZizka committed:
Fix typo Hiberanate

pmuir committed:
Add kitchensink-ml to root pom

pmuir committed:
typo

pmuir committed:
add kitchensink-ml-ear and cdi-portable extension to root pom

pmuir committed:
Move inter-app to complex-dependencies (thanks to Kabir Khan)

csadilek committed:
clarifications and javadoc improvements for the servlet-async quickstart

Commit:
Merge of changes from the latest updates on AeroGear to M5

sgilda committed:
Add target product (https://issues.jboss.org/browse/JDF-108)

rafabene committed:
Awestruct does NOT paset #{...} expressions

sgilda committed:
Fix incorrect product targets

bleathem commented:
Ok, pull request updated taking into account pmuir's comment's.
Also renamed helloworld-jsf to helloworld-rf with this pull request.

pmuir commented:
Looks ok to me.

sgilda commented:
Brian, I'll take care of this after you fix the connection-url problem.

bleathem commented:
Sande, I've update the PR with the fix to the JDBC connection URL problem.

sgilda commented:
Brian, there are 2 commits, both with the exact same fix. How did you do that? ;-)

Thanks! I'll try to look at it today.

bleathem commented:
That happens when I make the commit to the RichFaces repo, then merge it into the JSF repo.  I'm open to suggestions of an improved workflow, but that's what I've come up with so far.

LightGuard committed:
Fixing JDF-78

":Author:" is a keyword "Author:" is not :)

sgilda commented:
Hi Brian,
I finally got back to this today. I run into an error when I attempt to build it (mvn clean package):

[ERROR] Failed to execute goal on project jboss-as-kitchensink-rf: Could not resolve dependencies for project org.jboss.as.quickstarts:jboss-as-kitchensink-rf:war:7.1.2-SNAPSHOT: Could not find artifact com.sun.faces:jsf-impl:jar:2.1.5-jbossorg-1 in central (http://repo.maven.apache.org/maven2) -> [Help 1]

Thanks,
Sande

sgilda commented:
I forgot to mention, I don't see a anything with -jbossorg-1 in central. Would that be defined in our BOM?

bleathem commented:
That dependency is introduced in the pom with:

```
<!-- Import the JSF impl, to allow for application debugging -->
<dependency>
    <groupId>com.sun.faces</groupId>
    <artifactId>jsf-impl</artifactId>
    <scope>provided</scope>
</dependency>
```

The version is managed by the RichFaces BOM.  I guess the simplest thing to do is remove it?  It _is_ however quite useful...

Maybe I could put it in a "debug" profile.  Then the default build would run fine, and one still has the ability to inform the  IDE what JSF version to debug against.

WDYT?

sgilda commented:
I have to defer to Pete on this one. You're over my head. :-)
@pmuir : As Brian said, WDYT?

pmuir commented:
@bleathem issue is that we try to only use Maven central for dependencies, and this one obviously isn't in central. debug profile sounds reasonable to me, also making sure we switch back to the official version soon.

bleathem commented:
Ok, I'll add the debug dependency to a debug profile.  We track the JBoss AS jsf-impl version, and I don't believe they intend to stop forking the upstream impl (sometimes it's just a frozen SNAPSHOT).

bleathem committed:
Merge branch richfaces-master: renaming helloworld-jsf to helloworld-rf, and introducing kitchensink-rf

bleathem commented:
I created the "debug" profile.  

My successive merges had created quite a mess, so I through away my clone and re-merged the richfaces and jdf repositories with a single merge, force-pushing the result to my fork.  The result is a much cleaner commit history and I'm pretty sure the result is the same.

sgilda commented:
This builds, deploys, and runs great.
Pete, are you OK with this one now too?

pmuir commented:
Yes, please merge it :-)

On 20 Sep 2012, at 18:47, sgilda wrote:

> This builds, deploys, and runs great.
> Pete, are you OK with this one now too?
> 
> —
> Reply to this email directly or view it on GitHub.
